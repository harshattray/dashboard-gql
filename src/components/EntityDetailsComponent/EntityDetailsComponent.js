/**
 * @Author: harsha
 * @Date:   2019-05-17T03:21:48+05:30
 * @Last modified by:   harsha
 * @Last modified time: 2019-05-20T14:35:36+05:30
 */
import React, { Fragment, Component } from "react";
import { Button, Divider, Form, Grid, Segment, Card } from "semantic-ui-react";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import { createTasksAction } from "../../actions/TaskActions";
import FinancialDetailsComponent from "../FinancialDetailsComponent/FinancialDetailsComponent";
import SubmitDetailsForms from "../DetailsForm/DetailsForm";

class EntityDetailsComponent extends Component {
  componentDidMount() {
    // this.props.createTasksAction();
  }
  render() {
    const { financialDetailsStack } = this.props;
    return (
      <Fragment>
        <Segment placeholder>
          <Grid columns={3} relaxed="very" stackable>
            <Grid.Column>
              <Card
                color="teal"
                image="https://www.drivespark.com/car-image/640x480x100/car/300x225x41244514-bmw_3_series_gran_turismo.jpg.pagespeed.ic.ll0Wst0Uak.jpg"
              />
            </Grid.Column>
            <Grid.Column>
              <SubmitDetailsForms />
            </Grid.Column>
            <Grid.Column verticalAlign="middle">
              <FinancialDetailsComponent
                financialDetailsStack={financialDetailsStack}
              />
            </Grid.Column>
          </Grid>
        </Segment>
      </Fragment>
    );
  }
}

function mapStateToProps({ carStack }) {
  return {
    financialDetailsStack: carStack.financialDetails
  };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators({ createTasksAction }, dispatch);
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EntityDetailsComponent);
