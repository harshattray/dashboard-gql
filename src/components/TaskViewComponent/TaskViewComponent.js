/**
 * @Author: harsha
 * @Date:   2019-05-19T13:49:45+05:30
 * @Last modified by:   harsha
 * @Last modified time: 2019-05-20T20:20:18+05:30
 */

import React, { Fragment, Component } from "react";
import { Form, Card, Button, Segment, Dimmer, Loader } from "semantic-ui-react";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import { reduxForm, Field, formValueSelector, reset } from "redux-form";
import { fetchTasks, openModal } from "../../actions/TaskActions";
import CreateTasksModalComponent from "../CreateTasksModalComponent/CreateTasksModalComponent";
import TaskListComponent from "../TaskListComponent/TaskListComponent";

export class TaskViewComponent extends Component {
  componentDidMount() {
    this.props.fetchTasks(this.props.carId);
  }
  render() {
    const { isFetchingTasks, tasksInfo, openModal } = this.props;
    return (
      <Fragment>
        {isFetchingTasks ? (
          <Segment>
            <Dimmer active inverted>
              <Loader size="small">Loading</Loader>
            </Dimmer>
          </Segment>
        ) : (
          <Card>
            <Card.Content header="A list of tasks" />
            <Card.Content>
              {tasksInfo.tasks.length > 0 ? (
                <TaskListComponent />
              ) : (
                <p>No tasks found</p>
              )}
            </Card.Content>
            <Card.Content extra>
              <Button basic color="green" onClick={openModal}>
                Create Tasks
              </Button>
            </Card.Content>
            <CreateTasksModalComponent />
          </Card>
        )}
      </Fragment>
    );
  }
}

function mapStateToProps({ carStack, form, tasksStack }) {
  return {
    carId: carStack.carId,
    tasksInfo: tasksStack.tasksList,
    isFetchingTasks: tasksStack.isFetchingTasks
  };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators({ fetchTasks, openModal }, dispatch);
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(TaskViewComponent);
