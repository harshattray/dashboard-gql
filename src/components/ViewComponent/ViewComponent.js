/**
 * @Author: harsha
 * @Date:   2019-05-17T01:19:53+05:30
 * @Last modified by:   harsha
 * @Last modified time: 2019-05-20T05:03:07+05:30
 */
import React, { Component, Fragment } from "react";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import { ApolloClient } from "apollo-client";
import { createHttpLink } from "apollo-link-http";
import { InMemoryCache } from "apollo-cache-inmemory";
import gql from "graphql-tag";
import { getDetails } from "../../actions/EntityDetailsActions";
import EntityDetailsComponent from "../EntityDetailsComponent/EntityDetailsComponent";
import TaskViewComponent from "../TaskViewComponent/TaskViewComponent";
import { Dimmer, Loader, Segment, Card } from "semantic-ui-react";

class ViewComponent extends Component {
  componentDidMount() {
    this.props.getDetails();
  }
  render() {
    const { isFetching, carDataStack } = this.props;
    return (
      <Fragment>
        {isFetching ? (
          <Segment>
            <Dimmer active inverted>
              <Loader size="huge">Loading</Loader>
            </Dimmer>
          </Segment>
        ) : (
          <Card.Group>
            <Card
              fluid
              color="yellow"
              header={`${carDataStack.carStackDetails.make}${
                carDataStack.carStackDetails.model
              }`}
            />
            <EntityDetailsComponent />
            <TaskViewComponent />
          </Card.Group>
        )}
      </Fragment>
    );
  }
}

function mapStateToProps({ carStack }) {
  return {
    carDataStack: carStack,
    isFetching: carStack.isFetching
  };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators({ getDetails }, dispatch);
}
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ViewComponent);
