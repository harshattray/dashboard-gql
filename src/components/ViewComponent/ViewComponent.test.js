/**
 * @Author: harsha
 * @Date:   2019-05-20T16:52:29+05:30
 * @Last modified by:   harsha
 * @Last modified time: 2019-05-20T23:03:29+05:30
 */

import { shallow } from "enzyme";
import React from "react";
import { ViewComponent } from "./ViewComponent";
import EntityDetailsComponent from "../EntityDetailsComponent/EntityDetailsComponent";
import TaskViewComponent from "../TaskViewComponent/TaskViewComponent";

describe("Renders ViewComponent", () => {
  let wrapper;
  const props = {
    isFetching: false,
    getDetails: jest.fn(),
    carDataStack: {
      carStackDetails: {
        make: "BMW",
        model: "3-series"
      }
    }
  };

  beforeEach(() => {
    wrapper = shallow(<ViewComponent {...props} />);
  });

  it("expect to render View Component", () => {
    expect(shallow(<ViewComponent {...props} />)).toMatchSnapshot();
  });

  it("expect to contain EntityDetailsComponent ", () => {
    expect(wrapper.find(EntityDetailsComponent)).toMatchSnapshot();
  });

  it("renders EntityDetailsComponent ", () => {
    expect(wrapper.find(EntityDetailsComponent).exists()).toBe(true);
  });

  it("expect to contain TaskViewComponent ", () => {
    expect(wrapper.find(TaskViewComponent)).toMatchSnapshot();
  });
});
