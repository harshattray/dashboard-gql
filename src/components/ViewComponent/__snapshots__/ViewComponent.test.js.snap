// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders ViewComponent expect to contain EntityDetailsComponent  1`] = `ShallowWrapper {}`;

exports[`Renders ViewComponent expect to contain TaskViewComponent  1`] = `ShallowWrapper {}`;

exports[`Renders ViewComponent expect to render View Component 1`] = `ShallowWrapper {}`;
